Patch from Javier <jevv.cr@gmail.com> to fix http://bugs.sourcemage.org/show_bug.cgi?id=14448 (compilation with GCC 4.3.0)
--
diff -Naur STLport-5.1.5-orig/build/Makefiles/gmake/gcc.mak STLport-5.1.5/build/Makefiles/gmake/gcc.mak
--- STLport-5.1.5-orig/build/Makefiles/gmake/gcc.mak	2008-04-11 21:41:32.000000000 -0600
+++ STLport-5.1.5/build/Makefiles/gmake/gcc.mak	2008-04-11 22:00:48.000000000 -0600
@@ -29,6 +29,11 @@
 CXX_VERSION := $(shell ${CXX} --version | grep GCC | awk '{ print $$3; }')
 
 ifeq ($(CXX_VERSION),)
+# Current g++-4.3
+CXX_VERSION := $(shell ${CXX} --version | grep '^g++' | awk '{ print $$4; }')
+endif
+
+ifeq ($(CXX_VERSION),)
 # 2.95 report only version
 CXX_VERSION := $(shell ${CXX} --version)
 endif
diff -Naur STLport-5.1.5-orig/stlport/limits.h STLport-5.1.5/stlport/limits.h
--- STLport-5.1.5-orig/stlport/limits.h	2008-04-11 21:41:32.000000000 -0600
+++ STLport-5.1.5/stlport/limits.h	2008-04-11 21:39:51.000000000 -0600
@@ -24,7 +24,11 @@
 struct _exception;
 #endif
 
-#include _STLP_NATIVE_C_HEADER(limits.h)
+#if defined (_STLP_HAS_INCLUDE_NEXT)
+#  include_next <limits.h>
+#else
+#  include _STLP_NATIVE_C_HEADER(limits.h)
+#endif
 
 #if (_STLP_OUTERMOST_HEADER_ID == 0x201)
 #  if ! defined (_STLP_DONT_POP_HEADER_ID)
diff -Naur STLport-5.1.5-orig/stlport/stl/config/_gcc.h STLport-5.1.5/stlport/stl/config/_gcc.h
--- STLport-5.1.5-orig/stlport/stl/config/_gcc.h	2008-04-11 21:41:32.000000000 -0600
+++ STLport-5.1.5/stlport/stl/config/_gcc.h	2008-04-11 22:04:17.000000000 -0600
@@ -4,6 +4,8 @@
 
 #define _STLP_COMPILER "gcc"
 
+#define _STLP_HAS_INCLUDE_NEXT 1
+
 /* Systems having GLIBC installed have different traits */
 #if defined (__linux__)
 #  ifndef _STLP_USE_GLIBC

